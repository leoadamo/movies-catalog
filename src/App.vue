<template>
  <div
    class="w-[1180px] max-w-full min-h-screen py-8 mx-auto flex flex-col"
    data-testid="app"
  >
    <router-view v-slot="{ Component }">
      <transition name="fade" mode="out-in" appear>
        <component :is="Component" />
      </transition>
    </router-view>
  </div>
</template>
