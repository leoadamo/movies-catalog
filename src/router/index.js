// DEPENDENCIES
import { createRouter, createWebHistory } from "vue-router";

const routes = [
  {
    path: "/",
    name: "home",
    component: () => import("@/views/HomePage.vue"),
  },
  {
    path: "/movies/:id/:slug",
    name: "movie-detail",
    component: () => import("@/views/MovieDetail.vue"),
    props: true,
    beforeEnter: async (to, from) => {
      const store = await import("@/stores/movies");

      const { getMovieById } = store.useMoviesStore();

      const exists = getMovieById(to.params.id);

      if (!exists) {
        return {
          name: "not-found",
          params: { pathMatch: to.path.split("/").slice(1) },
          query: to.query,
          hash: to.hash,
        };
      }
    },
  },
  {
    path: "/:pathMatch(.*)*",
    name: "not-found",
    component: () => import("@/views/NotFound.vue"),
  },
];

const router = createRouter({
  history: createWebHistory(),
  routes,
});

export default router;
