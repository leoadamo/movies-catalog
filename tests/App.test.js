// DEPENDENCIES
import { describe, it, expect } from "vitest";
import { render } from "@testing-library/vue";

// ROUTER
import router from "@/router";

// COMPONENTS
import App from "@/App.vue";

describe("App.vue", () => {
  it("Should render the App", () => {
    const wrapper = render(App, {
      global: {
        plugins: [router],
      },
    });

    const app = wrapper.queryByTestId("app");

    expect(app).toBeTruthy();
    expect(app).toBeInTheDocument();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
