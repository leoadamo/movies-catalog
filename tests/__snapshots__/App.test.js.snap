// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > Should render the App 1`] = `
"<div class="w-[1180px] max-w-full min-h-screen py-8 mx-auto flex flex-col" data-testid="app">
  <transition-stub name="fade" mode="out-in" appear="true" persisted="false" css="true">
    <!---->
  </transition-stub>
</div>"
`;
